src/main/java/com/upsolver/datasources/jdbc/JDBCDataSource.java

```diff
@@ -63,6 +67,17 @@ public int getMaxShards() {
     public void setProperties(Map<String, String> properties) {
         ds = new HikariDataSource();
         String connectionString = properties.get(connectionStringProp);
+        String connectionProperties = properties.get(connectionPropertiesProp);
+        if (!connectionProperties.isBlank()) {
+            Properties props = new Properties();
+            try {
+                props.load(new StringReader(connectionProperties));
+                ds.setDataSourceProperties(props);
+            } catch (IOException e) {
+                logger.error("Unable to parse connection properties", e);
+                throw new RuntimeException("Unable to parse connection properties: '" + connectionProperties + "'", e);
```

Comment:
You have validate before this, so it shouldn't not parse at this point

Reply:
Yeah, but java wants me to throw only RuntimeException or to add this Exception to list to the API :(